examples/run_mock_client_main.cc

```diff
@@ -2,52 +2,136 @@
 #include <memory>
 #include <string>
 
+#include "google/protobuf/timestamp.pb.h"
 #include "grpcpp/grpcpp.h"
+#include "src/common/protocol_client/chunk_server_service_client.h"
 #include "src/common/protocol_client/master_metadata_service_client.h"
+#include "src/common/system_logger.h"
+#include "src/protos/grpc/chunk_server_file_service.grpc.pb.h"
+#include "src/protos/grpc/chunk_server_lease_service.grpc.pb.h"
 #include "src/protos/grpc/master_metadata_service.grpc.pb.h"
 
+using gfs::service::ChunkServerServiceClient;
 using gfs::service::MasterMetadataServiceClient;
 using google::protobuf::util::Status;
 using google::protobuf::util::StatusOr;
+using protos::grpc::AdvanceFileChunkVersionRequest;
 using protos::grpc::DeleteFileRequest;
-using protos::grpc::OpenFileReply;
+using protos::grpc::GrantLeaseRequest;
+using protos::grpc::InitFileChunkRequest;
 using protos::grpc::OpenFileRequest;
+using protos::grpc::ReadFileChunkRequest;
+using protos::grpc::RevokeLeaseRequest;
+using protos::grpc::WriteFileChunkRequest;
+
+template <typename T, typename U>
+void LogRequestAndResponse(T request, StatusOr<U> reply_or) {
+  LOG(INFO) << "Request sent: \n" << request.DebugString();
+  if (reply_or.ok()) {
+    LOG(INFO) << "Response received: \n" << reply_or.ValueOrDie().DebugString();
+  } else {
+    LOG(ERROR) << "Request failed: \n" << reply_or.status().ToString();
+  }
+}
 
 int main(int argc, char** argv) {
```

Comment:
It is probably useful to leave a top-level comment here clarifying that the code shown in this file is not only for a GFS-client to set up communications, but for all services that act like a client. For example when metadata asks the chunkserver to initialize a chunk, it takes a client role.

Reply:
Renamed the file to `run_mock_protocol_client_main`